src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -799,6 +799,35 @@ private bool IsOpenable(bool needToUncompress, bool needToLoadIntoMemory, out st
 
         private bool SizesTooLarge() => _compressedSize > uint.MaxValue || _uncompressedSize > uint.MaxValue;
 
+        private static CompressionLevel? MapCompressionLevel(BitFlagValues generalPurposeBitFlag)
+        {
+            // Information about the Deflate compression option is stored in bits 1 and 2 of the general purpose bit flags.
+            int deflateCompressionOption = (int)generalPurposeBitFlag & 0x6;
+
+            return deflateCompressionOption switch
+            {
+                0 => CompressionLevel.Optimal,
+                2 => CompressionLevel.SmallestSize,
+                4 => CompressionLevel.Fastest,
+                6 => CompressionLevel.NoCompression,
+                _ => null
+            };
+        }
+
+        private static BitFlagValues MapDeflateCompressionOption(BitFlagValues generalPurposeBitFlag, CompressionLevel? compressionLevel)
+        {
+            ushort deflateCompressionOptions = compressionLevel switch
```

Comment:
We need to avoid setting these two bits if the compression algorithm is _not_ DEFLATE, as described in the spec.